File: day1/puzzle2.py
""" Task: Read in a series of newline seperated values from a file. These values are deltas, and the initial
	value is 0. What is the first value that is reached twice?
"""
from itertools import cycle

def main():
    deltas = list()
    with open("input.txt") as f:
        currentvalue = 0
        for line in f:
            deltas.append(int(line))
    
    # Cycle through deltas and values
    currentvalue = 0
    pastvalues = set([currentvalue])
    for delta in cycle(deltas):
        currentvalue += delta 
        if currentvalue in pastvalues:
            break 
        else:
            pastvalues.add(currentvalue)
    
    print(currentvalue)



if __name__ == "__main__":
    main()

File: day1/puzzle2.jl
""" Task: Read in a series of newline seperated values from a file. These values are deltas, and the initial
	value is 0. What is the first value that is reached twice?
"""

function main()
    deltas = open("input.txt", "r") do file 
        deltalist = Int[];
        for line in eachline(file)
            push!(deltalist, parse(Int, line))
        end
        deltalist
    end
    
    # Loop through repeatedly keeping track of current and previous values
    currentvalue = 0
    pastvalues = Set([currentvalue])
    for delta in Iterators.cycle(deltas)
        currentvalue += delta 
        if currentvalue in pastvalues 
            break
        else
            push!(pastvalues, currentvalue)
        end
    end
    println(currentvalue)
end

main()
